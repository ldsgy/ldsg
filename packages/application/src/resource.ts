import { Resource } from "@ldsg/resource";
import express, { Express } from "express";
import { ApplicationSpecificResourceSettings } from "./types";

type CreateApp = () => Promise<Express>;

interface AddRoutesParams {
  app: Express;
}

type AddRoutes = (params: AddRoutesParams) => Promise<void>;

export class ApplicationResource extends Resource<ApplicationSpecificResourceSettings> {
  addRoutes: AddRoutes = async (params) => {
    const { app } = params;

    const {
      settings: { name },
      getFilteredResources,
    } = this;

    const getFilteredResourcesRes = getFilteredResources({});

    const { resources } = getFilteredResourcesRes;

    resources.forEach((resource) => {
      if ("addRoute" in resource && typeof resource.addRoute === "function") {
        resource?.addRoute();
      }
    });

    app.get("/", (req, res) => {
      res.send(`Hello, ${name ?? "World"}!`);
    });
  };

  createApp: CreateApp = async () => {
    const app = express();

    await this.addRoutes({
      app,
    });

    return app;
  };
}
