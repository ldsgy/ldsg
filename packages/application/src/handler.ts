import { Handler } from "@ldsg/handler";
import { Manifest } from "@ldsg/resource";
import { ApplicationResource } from "./resource";
import { ApplicationResourceSettings } from "./types";

export const handler: Handler<
  [Manifest.Resource<ApplicationResourceSettings>]
> = (params) => {
  return new ApplicationResource(params);
};
