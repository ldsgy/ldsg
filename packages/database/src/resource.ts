import { Resource, ResourceConstructorParams } from "@ldsg/resource";
import mongoose from "mongoose";
import { DatabaseSpecificResourceSettings } from "./types";

export class DatabaseResource extends Resource<DatabaseSpecificResourceSettings> {
  constructor(
    params: ResourceConstructorParams<DatabaseSpecificResourceSettings>
  ) {
    super(params);

    const { connect } = this;

    connect();
  }

  connect = () => {
    const {
      settings: { uri, connectOptions },
    } = this;

    return mongoose.connect(uri, connectOptions);
  };

  disconnect = () => {
    return mongoose.disconnect();
  };
}
