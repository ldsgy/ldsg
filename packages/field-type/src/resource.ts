import { Resource } from "@ldsg/resource";
import { FieldTypeSpecificResourceSettings } from "./types";

export class FieldTypeResource extends Resource<FieldTypeSpecificResourceSettings> {}
