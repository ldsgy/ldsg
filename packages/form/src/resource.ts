import { ObjectInfo, ObjectResource } from "@ldsg/object";
import { Resource } from "@ldsg/resource";
import { FormSpecificResourceSettings } from "./types";

interface GetFormInfoParams {
  /**
   * Platform
   * Such as mongoose\formily.
   */
  platform: string;
}

export interface FormInfo {
  /**
   * Form Name
   */
  name: string;

  /**
   * Input Object Info
   */
  inputObjectInfo: ObjectInfo;

  /**
   * Output Object Info
   */
  outputObjectInfo: ObjectInfo;
}

type GetFormInfo = (params: GetFormInfoParams) => FormInfo;

export class FormResource extends Resource<FormSpecificResourceSettings> {
  getFormInfo: GetFormInfo = (params) => {
    const { platform } = params;

    const {
      settings: { name },
      getResourcesFromSettings,
    } = this;

    const { inputObjectResource, outputObjectResource, workflowResource } =
      getResourcesFromSettings();

    const inputObjectInfo = (
      inputObjectResource as ObjectResource
    ).getObjectInfo({ platform });

    const outputObjectInfo = (
      outputObjectResource as ObjectResource
    ).getObjectInfo({ platform });

    const res: FormInfo = {
      name,
      inputObjectInfo,
      outputObjectInfo,
    };

    return res;
  };
}
