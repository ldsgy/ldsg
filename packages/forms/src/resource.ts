import { FormResource } from "@ldsg/form";
import { Resource } from "@ldsg/resource";
import { FormsSpecificResourceSettings } from "./types";

export class FormsResource extends Resource<FormsSpecificResourceSettings> {
  getSubFormResources = () => {
    const { id, getFilteredResources } = this;

    const { resources } = getFilteredResources({
      parentId: id,
    });

    return resources as FormResource[];
  };
}
