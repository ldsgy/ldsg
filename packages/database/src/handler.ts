import { Handler } from "@ldsg/handler";
import {
  InstantiateResourceParams,
  InstantiateResourceRes,
} from "@ldsg/resource";
import { instantiateResource } from "./instantiate-resource";
import { DatabaseResource } from "./resource";
import { DatabaseSpecificResourceSettings } from "./types";

export const handler: Handler<
  [InstantiateResourceParams<DatabaseSpecificResourceSettings>],
  InstantiateResourceRes<DatabaseResource>
> = instantiateResource;
