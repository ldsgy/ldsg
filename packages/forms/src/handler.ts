import { Handler, HandlerExtendedResourceSettings } from "@ldsg/handler";
import {
  InstantiateResourceParams,
  InstantiateResourceRes,
} from "@ldsg/resource";
import { instantiateResource } from "./instantiate-resource";
import { FormResource } from "./resource";
import { FormSpecificResourceSettings } from "./types";

export const handler: Handler<
  [
    InstantiateResourceParams<
      HandlerExtendedResourceSettings<FormSpecificResourceSettings>
    >
  ],
  InstantiateResourceRes<FormResource>
> = instantiateResource;
