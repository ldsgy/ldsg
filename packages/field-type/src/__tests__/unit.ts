import { HandlerResource } from "@ldsg/handler";
import { FieldTypeResource } from "../resource";
import {
  stringFieldTypeHandlerResourceRecord,
  stringFieldTypeResourceRecord,
} from "../resource-records";

test("field type", () => {
  new HandlerResource(stringFieldTypeHandlerResourceRecord);

  const fieldTypeResource = new FieldTypeResource(
    stringFieldTypeResourceRecord
  );

  const { fieldTypeInfo } = fieldTypeResource.getFieldTypeInfo({
    fieldProperties: {
      max: "10",
    },
  });

  expect(fieldTypeInfo).toMatchSnapshot();
});
