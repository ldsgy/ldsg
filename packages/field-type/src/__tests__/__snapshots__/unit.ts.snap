// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`field type 1`] = `
{
  "platformToTypeMap": {
    "COMMON": {
      "max": "10",
      "type": "string",
    },
    "GRAPHQL": undefined,
    "JSON": undefined,
    "MONGOOSE": undefined,
  },
}
`;
